src/main/java/gov/va/ascent/framework/audit/AuditLogger.java

```diff
@@ -4,83 +4,137 @@
-        MDC.put("tokenId", auditable.getTokenId());
-    }
+	static final Logger LOGGER = LoggerFactory.getLogger(AuditLogger.class);
+
+	/** Maximum length we are allowing for the "message" part of the log, leaving room for AuditEventData and JSON formatting */
+	private static final int MAX_MSG_LEN = 14336;
+	/** The string to prepend when a message must be split */
+	private static final String PREPEND = "SPLIT LOG SEQ#";
+
+	/*
+	 * private constructor
+	 */
+	private AuditLogger() {
+
+	}
+
+	/**
+	 * Debug.
+	 *
+	 * @param auditable the auditable
+	 * @param activityDetail the activity detail
+	 */
+	public static void debug(AuditEventData auditable, String activityDetail) {
+		addMdcSecurityEntries(auditable);
+		logMessage(null, SanitizationUtil.stripXSS(activityDetail));
+		MDC.clear();
+	}
+
+	/**
+	 * Info.
+	 *
+	 * @param auditable the auditable
+	 * @param activityDetail the activity detail
+	 */
+	public static void info(AuditEventData auditable, String activityDetail) {
+		addMdcSecurityEntries(auditable);
+		logMessage(MessageSeverity.INFO, SanitizationUtil.stripXSS(activityDetail));
+		MDC.clear();
+
+	}
+
+	/**
+	 * Warn.
+	 *
+	 * @param auditable the auditable
+	 * @param activityDetail the activity detail
+	 */
+	public static void warn(AuditEventData auditable, String activityDetail) {
+		addMdcSecurityEntries(auditable);
+		logMessage(MessageSeverity.WARN, SanitizationUtil.stripXSS(activityDetail));
+		MDC.clear();
+
+	}
+
+	/**
+	 * Error.
+	 *
+	 * @param auditable the auditable
+	 * @param activityDetail the activity detail
+	 */
+	public static void error(AuditEventData auditable, String activityDetail) {
+		addMdcSecurityEntries(auditable);
+		logMessage(MessageSeverity.ERROR, SanitizationUtil.stripXSS(activityDetail));
+		MDC.clear();
+
+	}
+
+	/**
+	 * Adds the MDC security entries.
+	 *
+	 * @param auditable the auditable
+	 */
+	private static void addMdcSecurityEntries(AuditEventData auditable) {
+		MDC.put("logType", "auditlogs");
+		MDC.put("activity", auditable.getActivity());
+		MDC.put("event", auditable.getEvent().name());
+		MDC.put("audit_class", auditable.getAuditClass());
+		MDC.put("user", auditable.getUser());
+		MDC.put("tokenId", auditable.getTokenId());
+	}
+
+	/**
+	 * Perform actions to split logs into 14 KB chunks and log separately.
+	 * A 14 KB message leaves room for AuditEventData and JSON formatting.
+	 * <p>
+	 * This is a temporary measure, until a way is found to mitigate the 16 KB log size limit imposed by Docker.
+	 *
+	 * @param severity
+	 * @param message
+	 */
+	private static void logMessage(MessageSeverity severity, String message) {
+		if (message != null && message.length() <= MAX_MSG_LEN) {
+			doLog(severity, message);
+		} else {
+			// split message into 14 KB strings
+			String[] messages = message == null ? new String[] { "" } : message.split("(?<=\\G.{14336})");
+			int seq = 0;
+			for (String part : messages) {
+				part = PREPEND + ++seq + " " + part;
```

Comment:
Is it possible to make the sequence # its own JSON attribute in log message? I think that will make it easier to track in Kibana. Is there already a unique identifier for the log entry so we know which sequence #s go together?

Reply:
As far as I can see in Kibana, the AuditEventData never makes it there - at least it is not visible, and does not appear to be available to lucene searches. Not really sure where that info is intended to show up.